packages/flame/lib/src/components/position_component.dart

```diff
@@ -332,6 +342,33 @@ class PositionComponent extends Component
   /// The absolute center of the component.
   Vector2 get absoluteCenter => absolutePositionOfAnchor(Anchor.center);
 
+  /// Returns the angle formed by component's orientation vector and a vector
+  /// starting at component's absolute position and ending at [target]. This
+  /// angle is measured in clockwise direction. [target] should be in absolute/world
+  /// coordinate system.
+  ///
+  /// Uses [nativeAngle] to decide the orientation direction of the component.
+  /// See [lookAt] to make the component instantly rotate towards target.
+  ///
+  /// Note: If target coincides with the current component, then it is treated
+  /// as being north.
+  double angleTo(Vector2 target) {
+    return math.atan2(
```

Comment:
Can't you do `absoluteCenter.angleTo(target) - (nativeAngle + absoluteAngle)` here btw?
I think that it should check with the `absoluteCenter` instead of `absolutePosition` since it will be giving a bit strange angles from the edges. And also, what purpose does `- angle` have in the current calculation?

Reply:
> Can't you do `absoluteCenter.angleTo(target) - (nativeAngle + absoluteAngle)` here btw?

That would actually produce undesirable rotation if `anchor` is not `Anchor.center`. Think of a sprite component where the sprite is not actually centered. In such case one would set anchor to a custom value and would expect the angle calculation to be about that point.

Reply:
True true!

Reply:
> And also, what purpose does `- angle` have in the current calculation?

I subtracted it so that I can find the absoluteAngle of parent. But now that you've mentioned it, it does look wrong. Right now it is not actually returning the *angle to* target. Rather, it returns the value than `angle` should be set to, to make it look at the target.

Reply:
I guess just removing it would make the calculation correct?

Reply:
I think lookAt will have to be change to `angle += angleTo(target)` and some test will also have to be modified.

Reply:
Made these changes. Also updated the example.